### PR TITLE
Support LowCardinality nullability with join_use_nulls

### LINKED ISSUE #15101: join_use_nulls on LowCardinality type does not work
**Describe the bug**
When I use the `set join_use_nulls = 1;` setting, LowCardinality columns still show up with the default `""` value.

**How to reproduce**
* Which ClickHouse server version to use
20.6.5.8

* Queries to run that lead to unexpected result
```sql
create table left (
    id integer
) ENGINE = Memory();

create table right (
    id integer,
    batch LowCardinality(String)
) ENGINE = Memory();

insert into left (id) select * from system.numbers limit 10;
insert into right (id, batch) values (0, 'test'), (1, 'test'), (2, 'test');

set join_use_nulls = 1;
select * from left left outer join right on left.id = right.id;
```

Results:
```
0	0     test
1	1     test
2	2     test
3	null  ""
4	null  ""
5	null  ""
6	null  ""
7	null  ""
8	null  ""
9	null  ""
```

**Expected behavior**
```
0	0       test 
1	1       test 
2	2       test 
3	null	null 
4	null	null 
5	null	null 
6	null	null 
7	null	null 
8	null	null 
9	null	null 
```

### PR DESCRIPTION
**Pull request by vdimir**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):

- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Support LowCardinality nullability with `join_use_nulls`, close #15101

### CODE REVIEW

**Comment by vdimir** [comment]
Fuzz reproduced in master, https://github.com/ClickHouse/ClickHouse/issues/23351

File: src/Interpreters/join_common.cpp [hunk 1]
```diff
@@ -49,20 +49,55 @@ ColumnPtr changeLowCardinality(const ColumnPtr & column, const ColumnPtr & dst_s
...
+
     if (column.column)
-        column.column = makeNullable(column.column);
+    {
+        if (column.column->lowCardinality())
+        {
+            /// Convert nested to nullable, not LowCardinality itself
+            auto new_col = column.column->cloneEmpty();
+            new_col->insertRangeFrom(*column.column, 0, column.column->size());
+            /// How to call non-constant methon on ColumnPtr without a copy?
```
Comment (Avogar) [inline src/Interpreters/join_common.cpp:94]: `column->assumeMutable()-><method_name>(...)`

**Comment by Avogar** [comment]
@vdimir let's remove coping in convertColumnToNullableand merge it.

**Comment by vdimir** [comment]
> @vdimir let's remove coping in convertColumnToNullableand merge it.

[`fb88547` (#23237)](https://github.com/ClickHouse/ClickHouse/pull/23237/commits/fb88547a04e3cdc7fe3e032f154fa50c07adc542)

File: src/Interpreters/join_common.cpp (hunk 1 above)
Comment (vdimir) [reply src/Interpreters/join_common.cpp]: [`fb88547` (#23237)](https://github.com/ClickHouse/ClickHouse/pull/23237/commits/fb88547a04e3cdc7fe3e032f154fa50c07adc542)

**Review by Avogar — APPROVED** [review]

**Comment by vdimir** [comment]
Fuzzer reproduced on master

```
DROP TABLE IF EXISTS r;
DROP TABLE IF EXISTS l_lc;

CREATE TABLE r (x UInt32, lc String) ENGINE = Memory;
CREATE TABLE l_lc (x UInt32, lc LowCardinality(String)) ENGINE = Memory;

INSERT INTO r VALUES (0, 'str');
INSERT INTO r_lc VALUES (0, 'str');

SELECT
    l.lc,
    materialize(r.lc),
    l.lc
FROM l_lc AS l
RIGHT JOIN r USING (lc)
;
```

```
Logical error: 'Bad cast from type DB::ColumnLowCardinality to DB::ColumnString'
```

**Pull request merged**